src/agent/agent.c

```diff
@@ -652,6 +652,48 @@ static AgentUnitInfo *agent_ensure_unit_info(Agent *agent, const char *unit) {
         return info;
 }
 
+static int passthrough_agent_method_cb(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        _cleanup_systemd_request_ SystemdRequest *req = userdata;
+
+        if (sd_bus_message_is_method_error(m, NULL)) {
+                /* Forward error */
+                return sd_bus_reply_method_error(req->request_message, sd_bus_message_get_error(m));
+        }
+
+        _cleanup_sd_bus_message_ sd_bus_message *reply = NULL;
+        int r = sd_bus_message_new_method_return(req->request_message, &reply);
+        if (r < 0) {
+                return r;
+        }
+
+        r = sd_bus_message_copy(reply, m, true);
+        if (r < 0) {
+                return r;
+        }
+
+        return sd_bus_message_send(reply);
+}
+
+static int passthrough_agent_method(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
```

Comment:
The other API methods use the naming pattern `agent_method_<xyz>`, so this should be `agent_method_passthrough`. Maybe we could also specify the "target", e.g. `agent_method_passthrough_to_systemd` to be explicit?

Reply:
Done